### PR TITLE
feat(all): zero value to default value

### LINKED ISSUE #86: feat(component): treat zero value of configurations as default values
- If a component requires a lot of parameters, a lot of default values must be initialized in config; otherwise, **error** will be returned which makes the usage of the component harder.
- All of the related tests and examples must also be maintained following this change.
- The specific default value facility must be tested.

### PR DESCRIPTION
**Pull request by sahandsafizadeh**
1. **Feat**: Simplify usage by placing default values for _zero values_.
- Now for optimizers (`SGD`, `Adam` and `AdamW`) and `BatchNorm` layer, user can specify the hyper-parameters they need and the rest will be filled by default values. This way, while code is cleaner, component still continues to use its best settings.
- This is done where _zero value_ (mostly `0.` for `float64`) has no meaning. For instance, `Softmax` activation is changed to not have any initialization facility!
- Examples are updated following this changes.

2. **Fix**: remove broken `gosec` tool from `Makefile`.

### CODE REVIEW

**Review by sahandsafizadeh — COMMENTED** [review]
reviewed
